File: src/backend/CurrentBackend.ts
import { query } from "src/query";
import type { LoopEvent } from "src/types/events";
import type { AgentBackend, BackendQueryParams } from "./AgentBackend";

export class CurrentBackend implements AgentBackend {
  readonly name = "current" as const;

  async *query(params: BackendQueryParams): AsyncGenerator<LoopEvent, void> {
    yield* query(params);
  }
}

File: python/src/docker_agent/backend/current_backend.py
"""Current backend — delegates to the query() generator.

Parity: ``src/backend/CurrentBackend.ts``.
"""

from __future__ import annotations

from collections.abc import AsyncIterator

from docker_agent.backend.agent_backend import BackendQueryParams
from docker_agent.query import query
from docker_agent.types.events import LoopEvent


class CurrentBackend:
    name = "current"

    async def query(self, params: BackendQueryParams) -> AsyncIterator[LoopEvent]:
        async for ev in query(
            messages=params.messages,
            ctx=params.ctx,
            provider=params.provider,
            model=params.model,
        ):
            yield ev